airflow-ctl/src/airflowctl/ctl/commands/provider_commands.py

```diff
@@ -0,0 +1,39 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+import sys
+
+import rich
+
+from airflowctl.api.client import get_client
+
+
+def list(args) -> None:
```

Comment:
```suggestion
@provide_api_client
def list(args, cli_api_client) -> None:
```

https://github.com/apache/airflow/blob/319ad7207c143c61b05f8ef8a1772c633671f6d7/airflow-ctl/src/airflowctl/api/client.py#L241

This is a decorator you can directly mount where you don't need to check any credentials or client, etc... :)

Reply:
Also, we don't need to implement any command which have a operation. Simple ones like the `list` `get` `created` for all group commands. Others still need implementation if needed.

See this one. I will merge later this night. It is ready and working :)

https://github.com/apache/airflow/pull/48146